core/arch/arm/kernel/stmm_sp.c

```diff
@@ -232,102 +244,184 @@ static void uncompress_image(void *dst, size_t dst_size, void *src,
+	uint32_t hob_table_offset = 0;
+	void *guid_hob_data = NULL;
+
+	uncompressed_size_roundup = ROUNDUP(stmm_image_uncompressed_size,
+					    SMALL_PAGE_SIZE);
+	stmm_image_addr = sp_addr;
+	stmm_heap_addr = stmm_image_addr + uncompressed_size_roundup;
+	stmm_sec_buf_addr = stmm_heap_addr + stmm_heap_size;
+	hob_table_offset = sizeof(struct ffa_boot_info_header_1_1) +
+		sizeof(struct ffa_boot_info_1_1);
+
+	hob_table = efi_create_hob_list(sp_addr, sp_size,
+					stmm_sec_buf_addr + hob_table_offset,
+					stmm_sec_buf_size - hob_table_offset);
+	if (!hob_table) {
+		EMSG("Failed to create hob_table.");
+		return NULL;
+	}
+
+	ret = efi_create_fv_hob(hob_table, sp_addr, uncompressed_size_roundup);
+	if (ret) {
+		EMSG("Failed to create fv hob.");
+		return NULL;
+	}
+
+	ret = efi_create_guid_hob(hob_table, &ns_buf_guid,
+				  sizeof(struct efi_mmram_descriptor),
+				  &guid_hob_data);
+	if (ret) {
+		EMSG("Failed to create ns buffer hob.");
+		return NULL;
+	}
+
+	mmram_desc_data = guid_hob_data;
+	mmram_desc_data->physical_start = stmm_ns_comm_buf_addr;
+	mmram_desc_data->physical_size = stmm_ns_comm_buf_size;
+	mmram_desc_data->cpu_start = stmm_ns_comm_buf_addr;
+	mmram_desc_data->region_state = EFI_CACHEABLE | EFI_ALLOCATED;
+
+	mmram_resv_data_size = sizeof(struct efi_mmram_hob_descriptor_block) +
+			       sizeof(struct efi_mmram_descriptor) * 5;
+
+	ret = efi_create_guid_hob(hob_table, &mmram_resv_guid,
+				  mmram_resv_data_size, &guid_hob_data);
+	if (ret) {
+		EMSG("Failed to create mm range hob");
+		return NULL;
+	}
+
+	mmram_resv_data = guid_hob_data;
+	mmram_resv_data->number_of_mm_reserved_regions = 4;
+	mmram_desc_data = &mmram_resv_data->descriptor[0];
+
+	mmram_desc_data[0].physical_start = stmm_image_addr;
+	mmram_desc_data[0].physical_size = uncompressed_size_roundup;
+	mmram_desc_data[0].cpu_start = stmm_image_addr;
+	mmram_desc_data[0].region_state = EFI_CACHEABLE | EFI_ALLOCATED;
+
+	mmram_desc_data[1].physical_start = stmm_sec_buf_addr;
+	mmram_desc_data[1].physical_size = stmm_sec_buf_size;
+	mmram_desc_data[1].cpu_start = stmm_sec_buf_addr;
+	mmram_desc_data[1].region_state = EFI_CACHEABLE | EFI_ALLOCATED;
+
+	mmram_desc_data[2].physical_start = stmm_ns_comm_buf_addr;
+	mmram_desc_data[2].physical_size = stmm_ns_comm_buf_size;
+	mmram_desc_data[2].cpu_start = stmm_ns_comm_buf_addr;
+	mmram_desc_data[2].region_state = EFI_CACHEABLE | EFI_ALLOCATED;
+
+	mmram_desc_data[3].physical_start = stmm_heap_addr;
+	mmram_desc_data[3].physical_size = stmm_heap_size;
+	mmram_desc_data[3].cpu_start = stmm_heap_addr;
+	mmram_desc_data[3].region_state = EFI_CACHEABLE;
+
+	*hob_table_size = hob_table->efi_free_memory_bottom -
+			  (efi_physical_address_t)hob_table;
+
+	return hob_table;
+}
+
 static TEE_Result load_stmm(struct stmm_ctx *spc)
 {
-	struct stmm_boot_info *boot_info = NULL;
-	struct stmm_mp_info *mp_info = NULL;
-	TEE_Result res = TEE_SUCCESS;
+	struct ffa_boot_info_header_1_1 *ffa_boot_info_hdr = NULL;
```

Comment:
How about `hdr` instead of  `ffa_boot_info_hdr`, and `desc` instead of `ffa_boot_info_desc`?